### ISSUE TITLE
Fundamental misunderstanding of the Linux ecosystem.

### ISSUE THREAD

**Issue by Zambito1** [do-not-theme/do-not-theme.github.io#15]
https://github.com/do-not-theme/do-not-theme.github.io/blob/2dd083c49ce1c460486662c6b8ffcdbfb4bcbff0/index.html#L46

This line expresses that the intent of the letter is that it is a plea to **distributions** (or rather, their maintainers), and is not targeted at "tinkerers playing with their own setup". However, distribution maintainers and tinkerers playing with their own setup are one in the same. The only difference between the two is that a distribution maintainer makes their configuration available for others to choose to use (they quite literally **distribute** it).

Many desktop Linux users value the freedoms that define "Free Software" as outlined by the [FSF](https://www.gnu.org/philosophy/free-sw.html), and while "theming" is not necessarily software, users apply the same values (meaning they could modify and redistribute a theme as they wish; many themes are even licensed as GPL).

If you don't want people to theme your software (or rather, you don't want your software to be themed by distribution maintainers), why not write your own license forbidding just that? No one will want to touch it with a 10 foot pole, but hey, you won't have distribution maintainers theming your software.

**Comment by bertob** [comment]
Just because something is technically possible and not illegal doesn't mean it's not a dick move. If people want to still do it that's their business, we're just explaining why it's harmful for everyone involved, and nicely asking them not to.

**Comment by StarterX4** [comment]
They're like an arrogant chef who, after cooking your dinner, instructs the waiter to remove all the salt and pepper shakers from the table because they don't want you changing the flavor of their dish.

**Comment by bertob** [comment]
Well no, quite the opposite. To stick with your analogy, the problem is that the waiter is putting extra salt on everyone's food before they serve it, and the chef is tired of being blamed because the food is too salty :)

**Comment by mrmeszaros** [comment]
Still, I think this the argument holds some ground.

My first instinct finding the tag on an app's GitHub page, then reading the letter was:
_why do You want to make it harder for me (or a theme-dev's) life?_

I understand that theming can break the looks of an app. Also, developing an app that is themable is much harder than one that has it's own look and feel.

However, using a theme is for me (and to the linux-ricing community) an internal part of linux. This is also underlined by the freedesktop.org specifications to theming.

**Comment by Exalm** [comment]
What freedesktop specifications? :) The icon name spec that hasn't been updated since 2006 and that nobody follows?

> However, using a theme is for me (and to the linux-ricing community) an internal part of linux

That's your (and the linux ricing community's) problem. Don't make it others' problem.

If you don't like how something looks like or how others maintain their apps - feel free to fork them and do whatever you want in there. Don't expect others to do the work for you.

**Comment by mrmeszaros** [comment]
> What freedesktop specifications?

Like this: [Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/) - I see that this is quite old (2013 to be precise), and is on version `0.13`.
To contrast it, the [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/) has been updated in 2020 and is on version `1.5`.

> nobody follows

That is not true. Many apps use it (like inkscape can be themed eg. with Numix) and many apps do not use it (still understandable, as there is no 1.0 version).

However, I think it is a good starting point, and should not be dismissed so easily - and definitely not with such hostility.

> Don't make it others' problem.

I may have been a bit hostile myself in my tone and with reviving this dead thread. I would like to apologize for that.
* I do not want to make it other's problem.
* I do not expect others to do the work for me.

However, I do think, that having a way to customize the desktop experience - and to have a choice between options, is a good direction.

I also think that while Adwaita looks fine, there are other themes that also have a good design, like e.g.: [Orchis-theme](https://github.com/vinceliuice/Orchis-theme).

I know that GNOME and in fact most FOSS software is understaffed, and so work is slow and needs to sort the priorities - in which theming and related topics are low on the list.

#### But back to topic.

I do partially agree with the original issuer: if GNOME is _Free Software_ as FSF defines it, then applying themes is part of the freedom, and fighting against it - while is in any developers freedom - might cause some people to backlash and reconsider their options.

Again, I want to emphasize that I understand and know theming is not top priority, and every dev maintains their apps as they please (Yay, freedom!). Also, GNOME has the freedom, to remove theming or dismiss or ignore it.

However, since the code is open source, and uses CSS and some people like to tinker, there will be people trying to theme it. And they will share it most likely. Others will use. And they will complain. Sometimes to the theme-dev (rightfully) - but sometimes to You - the app-devs - to try to accomodate. Yes, it is (and will be) annoying, because extra work, and why would anyone want it to look different. But I digress.

I think the solution (long term) would be to work on specifications and extension points, even if on a minimal subset (like having redefinable color styles), then maybe opening up towards padding and margin-sizes or border-radii etc.

#### Anyway, TLDR
If I would have to summarize my point, I would say:

"please try to keep a conversation and do not shun others for wanting to theme the apps"

**Comment by Exalm** [comment]
> Like this: [Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/) - I see that this is quite old (2013 to be precise), and is on version 0.13.

And then you realize that everybody extends it. For example, there's no mention of symbolic icons there, which is what both GNOME and KDE use almost exclusively nowadays, and GNOME and KDE's implementations of symbolics are [not compatible](https://gitlab.gnome.org/GNOME/gtk/-/issues/1762) with each other:

![image](https://github.com/do-not-theme/do-not-theme.github.io/assets/1510457/7274a75f-030e-455e-8efd-c72726cfe0af)

 and they also have vastly different sets of icons, so if you run a KDE app in GNOME it will be missing half the icons and the other way around. In fact, looks like running KDE apps keep Breeze icons in GNOME now. Good, since they were basically unusable before.

It's still a mess in dark style though, since it doesn't change icons to breeze-dark which is required for cases like this:

![image](https://github.com/do-not-theme/do-not-theme.github.io/assets/1510457/bd47d05f-4f08-4251-acf8-d70d682b4621)

Meanwhile most parts of that spec are unused. So it's followed in only technical sense.

Besides, that spec is the main reason working with icons from apps is a royal pain (e.g. you have to follow that dir structure even though it makes 0 sense in a world where you only care about symbolic icons and bundle everything in apps), so at least in libadwaita we'll very likely stop supporting icon themes at all it in near future (they are still supported atm)

> However, I do think, that having a way to customize the desktop experience - and to have a choice between options, is a good direction.

Read https://ometer.com/preferences.html and https://blogs.gnome.org/tbernard/2018/10/15/restyling-apps-at-scale/

You can either have quality apps or themes. We chose the former and - surprise - we're getting tons of quality apps nowadays.

> I do partially agree with the original issuer: if GNOME is Free Software as FSF defines it, then applying themes is part of the freedom

Let's quote the definition you're referring to:

> The freedom to run the program as you wish means that you are not forbidden or stopped from making it run. This has nothing to do with what functionality the program has, whether it is technically capable of functioning in any given environment, or whether it is useful for any particular computing activity.

So no. This just tells me that you haven't read or understood it.

> I think the solution (long term) would be to work on specifications and extension points, even if on a minimal subset (like having redefinable color styles), then maybe opening up towards padding and margin-sizes or border-radii etc.

Oh, we've already solved it. Themes aren't a thing in newer GNOME apps, and this website is from 2019.

We still support `GTK_THEME` for now, but will stop as well in future, as it's causing problems like https://discourse.gnome.org/t/how-to-let-gtkmenubutton-be-flat-using-gtkbuilder/15229 and https://gitlab.gnome.org/GNOME/libadwaita/-/issues/677

Experience is showing that it very much was the right move, and the people who care about themes are a tiny tiny minority and will manage somehow.

**Comment by mrmeszaros** [comment]
Okay, first: wow, You are quick. Thanks :)
This is a lot to process, and I have work tomorrow, so it will likely take time to respond.

But to address the obvious:

#### Icon-Theme-Spec
Yep. The FreeDestkop Icon-Theme-Spec is old, has not reached 1.0 - as I addressed it. I might have misunderstood Your comment on:
> What freedesktop specifications?
I thought You are not aware of it. Clearly You are. And I agree with Your critique on it - it is only followed in a technical sense and since there are a lot of issues with application icons it's cost is definitely higher than what it brings to the table.

I sense Your frustration with the legacy we have with that.

#### Theming motivations
My main motive for theming is to have a mostly unified look.

Different icons styles, padding amounts, and colors increase the overall system UI complexity and make "clutter".
I want as little clutter as possible.
That was always my main motive behind theming - it helped to bring a unified look - with same colors / icons / graphical elements meaning same things.
This was probably also fueled by the older apps bad UI design. I turned to Material look and did not yet care check out GNOME HIG (before today last time I checked it was not really accessible).

Anyway, regarding visual clutter and UX/UI I think we are on the same page.
Even if we currently see different ways to achieve it.

#### FSF and Freedom

> > The freedom to run the program as you wish means that you are not forbidden or stopped from making it run. This has nothing to do with what functionality the program has, whether it is technically capable of functioning in any given environment, or whether it is useful for any particular computing activity.
>
> So no. This just tells me that you haven't read or understood it.

So I AM allowed to theme it however I can and like, am I not? To fiddle with it. To run the program as I wish. Even if it is not useful. Even if some buttons turn out white-on-white.
I have to warn You English is my third language, but I do think I understand it.

#### On GNOME dropping theming support
Well, part of me is glad that Adwaita looks nice (again, GNOME-HIG looks cool) and apps look better.
Then part of me has this itching feeling that it is going towards a one-size-fits all.

Again, thanks for the articles - they look interesting at a glance - I will read them, and maybe extend my response based on them.

**Comment by Exalm** [comment]
> Different icons styles, padding amounts, and colors increase the overall system UI complexity and make "clutter".
I want as little clutter as possible.

Cool, but you also get to maintain your system. Another blog post comes to mind: https://blogs.gnome.org/tbernard/2020/01/17/doing-things-that-scale/

> This was probably also fueled by the older apps bad UI design

Ultimately you can either fix an old app, or maintain tweaks forever locally. Your choice, but you don't get to ask to make it easy to work around problems when it significantly harms other areas. Or rather you can ask, of course, but don't be surprised when people doing the work say no.

> So I AM allowed to theme it however I can and like, am I not?

Nobody is prohibiting you from doing that. You have source code, you can modify it and run that. You probably don't want to because it's a lot of work, but it's a lot of work either way - the only difference is whether you're doing it or others.

What you're asking for is _not_ about being able to run things however you like, it's about having somebody provide a specific feature for you. There's a difference between being allowed to do X and having an easy option to enable X that somebody else conveniently provides and maintains for you.

**Comment by mrmeszaros** [comment]
> What you're asking for is not about being able to run things however you like, it's about having somebody provide a specific feature for you.

Not quite. I commented on this issue, because I think You (all who signed this) asking distributors not to work on their custom themes is - while understandable - overstapping a boundary.

Now, I think You and GNOME has every right to ensure software is maintainable.
Part of that is cutting back old features, lessening the burden of legacy.

What I am saying You asking distributors to cease and desist from using custom themes is kind of overstepping your boudaries.

It's their work - keeping, maintaining those styles.
Including applications in their distribution as they see fit.
Naming it ORC or GOBLIN and letting others use it.

I might be in the wrong but as I see - from your words - a person is free to do changes downstream (on their machine).
However, if they make it open to use or distribute, then it is wrong?

Ps.: I haven't finished reading all of the articles You linked, but they give good explanation to Your point of view.

**Comment by airtonix** [comment]
> we're just explaining why it's harmful for everyone involved, and nicely asking them not to.

except it's not harmful to anyone except your own ego.
